synapse/util/wheel_timer.py

```diff
@@ -11,17 +11,20 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import Generic, List, TypeVar
+from typing import Generic, Hashable, List, Set, TypeVar
 
-T = TypeVar("T")
+T = TypeVar("T", bound=Hashable)
 
 
 class _Entry(Generic[T]):
     __slots__ = ["end_key", "queue"]
 
     def __init__(self, end_key: int) -> None:
         self.end_key: int = end_key
-        self.queue: List[T] = []
+
+        # We use a set here as otherwise we can end up with a lot of duplicate
+        # entries.
+        self.queue: Set[T] = set()
```

Comment:
maybe `self.elements` instead of `self.queue`, since this is no longer ordered?

Reply:
Good idea, done!